### DISCUSSION TITLE
Spreadsheet Report From BookLibConnect List

### DISCUSSION THREAD

**Discussion by Sailfish** [audiamus/BookLibConnect#46] [General]
Is there a field on the Book Lib Connect panel(s) that indicates how many book titles were loaded? I was able to determine it, I believe?, by selecting the first title item on the list and then entering ctl-A to Select All then doing a ctl-C to Copy the list then doing a ctl-V to Paste in my text editor and scrolling to the end and, after subtracting 1 from the line number to account for the heading line.

**audiamus** prior reply:
> No, there are no statistics functions at the moment. But copying the entire table with Ctrl+A/Ctrl+C is pretty smart, I think. I wasn't even aware that the underlying Windows Forms control "DataGridView" supports this. You get a CSV-style output, with a tab as the column separator here instead of the standard comma. But any spreadsheet software will handle the import. The "state" column will remain empty, though, since it's an image column.

**Comment by Sailfish** [comment]
For completeness, the following is the process I used to create the spreadsheet in Excel.

1. From **BookLibConnect** list panel, **_click/tap_** the column name you want to sort the list
2. Select any row and enter **_ctrl-A_** to _Select All_
3. Enter **_ctrl-C_** to copy list to _Clipboard_
4. Open desired test editor, e.g. Notepad
5. Enter **_ctrl-V_** to paste _Clipboard_ list
6. Save the text file as _SomePath\SomeName.csv_
7. Open _SomeName.csv_ in **Excel**
8. Select **_all of column A_**
9. Select **_Data_** tab
10. Select **_Text to Columns_** toolbar button
11. Select **_Delimited_** radio button in dialog box then select **_Next_**
12. Select **_Tab_** checkbox is selected under **Delimiters** then select **_Finished_** button
13. Save as an Excel file.

**Reply by audiamus** [reply]
Well, at least with Libre Office it's even simpler, but this may work with Excel as well:

1. Copy the contents of the book list from Book Lib Connect to the clipboard, Ctrl+A, Ctrl+C, as suggested above.
2. Create a new empty spreadsheet in Libre Office Calc (the counterpart to Excel).
3. Paste the clipboard into the new spreadsheet, Ctrl+V.

That's it.

**Reply by Sailfish** [reply]
ah, yes, so it does. I'm used to dealing with comma-separated-value lists where, since "tab-separated" is the default delimiter setting, more fiddling is required beforehand.

Anyway, it's all good and this now allows me to easily generate a quick interim spreadsheet of my library.

Much appreciated.
